google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -107,9 +106,6 @@ public Void apply(WriteLogEntriesResponse input) {
 
   private static <I, O> Future<O> transform(Future<I> future,
       Function<? super I, ? extends O> function) {
-    if (future instanceof ListenableFuture) {
-      return Futures.transform((ListenableFuture<I>) future, function);
-    }
```

Comment:
This was done to keep the dynamic type an instance of `ListenableFuture` if the input future was already a `ListenableFuture`.

- `Futures.transform` returns a `ListenableFuture` created via [`AbstractTransformFuture.create`](https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/AbstractTransformFuture.java#L37) 
- `Futures.lazyTrasform` returns an instance of `Future`

This was introduced in https://github.com/GoogleCloudPlatform/google-cloud-java/pull/1168 to allow users to use `JdkFutureAdapters.listenInPoolThread` without requiring additional threads. More context can be found in https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1148